service/resource/deployment/deployment.go

```diff
@@ -19,6 +21,17 @@ const (
 	virtualNetworkSetupTemplate = "virtual_network_setup.json"
 )
 
+// keyVaultSecrets is used to pass secrets to Key Vault as a secure object.
```

Comment:
Here are new structs for passing the secrets data. It gets serialised to JSON when the deployment is created.

I like this and I think we can use it more to reduce the number of parameters we need to provide to the main template.

Reply:
FYI we used a separate `spec.go` file for such types in the past for better separation.

Reply:
I see this file is not for the resource implementation. Then this may just be good enough.

Reply:
I moved these to `spec.go` as yes I think that's tidier.